src/__tests__/comment.test.ts

```diff
@@ -15,32 +15,46 @@ describe(`comment`, () => {
     with: text(),
   });
 
-  const db = defineDb({ foo, bar }, () => Promise.resolve({ rows: [], affectedCount: 0 }));
+  let lastQuery: string;
+  let lastParameters: any[];
 
-  it(`select`, () => {
-    const query = db
+  const db = defineDb({ foo, bar }, (query, parameters) => {
+    lastQuery = query;
+    lastParameters = parameters;
+    return Promise.resolve({ rows: [], affectedCount: 0 });
+  });
+
+  const getLastQueryObj = () => {
+    return {
+      parameters: lastParameters,
+      text: lastQuery,
+    };
+  };
+
+  it(`select`, async () => {
+    await db
       .comment('This is a comment')
       .select(db.bar.id)
       .from(db.bar)
       .where(db.bar.name.eq('Test'));
 
-    expect(toSql(query)).toStrictEqual({
+    expect(getLastQueryObj()).toStrictEqual({
```

Comment:
Might be nice to check both `toSql` and the `await` thing.

Maybe with a helper, e.g.

```ts
const checkQueryAsync = (query, expected) => {
    const toSqlResult = ...;
    expect(toSqlResult).toEqual(expected);
    await query;
    expect(lastQuery).toEqual(expected);
};
```